### DISCUSSION TITLE
Identifying alpha channels across a large number of files (noob Windows user)

### DISCUSSION THREAD

**Discussion by veavibork** [ImageMagick/ImageMagick#2563] [Help] [answered]
I'm trying to sort ~18k images by whether they have transparency, whether they have "transparency" (255,255,255 being used as a blank background), or whether there's no transparency involved.

My google-fu is failing me. I've found references to making it work but following the same steps keeps kicking out a variety of errors. The only thing I've succeeded in doing is dumping verbose into a log, which results in 220 megabytes of text I can't readily work with... help?

**Comment by fmw42** [comment, marked as answer]
What is your exact command line?  What is your IM version? Do I understand that you want to test on transparent white only?

**Reply by veavibork** [reply]
Version is 7.0.10-29. I have three outcomes-

* transparent pixels in use
* 255,255,255 pixels in use
* neither above case are true

I don't need to get it done all in one pass, if I can get a list of files where transparent pixels are used and then a list of file where 255,255,255 black pixels are used that's enough for my needs.

I don't know the right command line. I've been trying variants of convert, identify, and mogrify, and while sometimes I get it working on one file I can't make it translate to all files.

**Reply by fmw42** [reply]
I still do not understand. What are 255,255,255 black pixels.  rgb=255,255,255 is white!

You can find any transparent images by

`magick image -channel a -separate +channel -format "%[fx:mean<1?1:0]" info:`

if the return value is 1, it has some transparency; otherwise, it has no transparency.

You can find if 255,255,255 by

`magick image -fill black +opaque white -format "%[fx:mean>0?1:0]" info:`

if the return value is 1, it has some white pixels; otherwise, it has no white pixels.

**Reply by veavibork** [reply]
Yikes! You're right - sorry, 0,0,0 rather than 255,255,255.

How do I apply this to a large number of files though? Adding mogrify to it promptly breaks the program-

magick mogrify 1012_wispout.png -channel a -separate +channel -format "%[fx:mean<1?1:0]" info:
mogrify: unable to open image '%[fx:mean<1?1:0]:': Invalid argument @ error/blob.c/OpenBlob/3537.
mogrify: unrecognized image format `%[fx:mean<1?1:0]' @ error/mogrify.c/MogrifyImageCommand/5062.

**Reply by fmw42** [reply]
You have to use a script loop over each image. Sorry, I am not a Windows user. I can do it in bash. But if you need .bat, then someone else will need to help.

**Reply by veavibork** [reply]
After some fiddling, I discovered that batch files absolutely _hate_ special characters, and escaping them is a giant pain in the tuchis. But I can loop like this right from command-

for %f in (.\0000.png) do magick %f -channel a -separate +channel -format "%[fx:mean<1?1:0]" info:

-and it works just like it should. Thanks!
